modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -974,9 +1388,9 @@ GAPI_FLUID_KERNEL(GFluidAbsDiffC, cv::gapi::core::GAbsDiffC, false)
         };
 
         //     DST     SRC     OP            __VA_ARGS__
-        UNARY_(uchar , uchar , run_arithm_s, dst, src, scalar, ARITHM_ABSDIFF);
-        UNARY_(ushort, ushort, run_arithm_s, dst, src, scalar, ARITHM_ABSDIFF);
-        UNARY_( short,  short, run_arithm_s, dst, src, scalar, ARITHM_ABSDIFF);
+        UNARY_(uchar , uchar , run_absdiffc, dst, src, scalar, ARITHM_ABSDIFF);
+        UNARY_(ushort, ushort, run_absdiffc, dst, src, scalar, ARITHM_ABSDIFF);
+        UNARY_( short,  short, run_absdiffc, dst, src, scalar, ARITHM_ABSDIFF);
```

Comment:
However this action leads necessity to create new `UNARY_` macro. How do you think it is necessary/reasonably?

Reply:
No, it does not... UNARY_ macro:
```
#define UNARY_(DST, SRC, OP, ...)                         \
    if (cv::DataType<DST>::depth == dst.meta().depth &&   \
        cv::DataType<SRC>::depth == src.meta().depth)     \
    {                                                     \
        GAPI_DbgAssert(dst.length() == src.length());       \
        GAPI_DbgAssert(dst.meta().chan == src.meta().chan); \
                                                          \
        OP<DST, SRC>(__VA_ARGS__);                        \
        return;                                           \
    }
```
So, it re-directs all the arguments after `OP` to `OP<DST, SRC>()`. So, if you delete the last argument from `run_absdiffc()`, there won't be necessity to write `ARITHM_ABSDIFF` here as well.

Reply:
Following output appears when I tried to remove last argument of the 'run_absdiffc():
akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1377): error C2672: 'run_absdiffc': no matching overloaded function found
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1377): error C2780: 'void cv::gapi::fluid::run_absdiffc(cv::gapi::fluid::Buffer &,const cv::gapi::fluid::View &,const float [])': expects 3 arguments - 4 provided
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1231): note: see declaration of 'cv::gapi::fluid::run_absdiffc'
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1378): error C2672: 'run_absdiffc': no matching overloaded function found
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1378): error C2780: 'void cv::gapi::fluid::run_absdiffc(cv::gapi::fluid::Buffer &,const cv::gapi::fluid::View &,const float [])': expects 3 arguments - 4 provided
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1231): note: see declaration of 'cv::gapi::fluid::run_absdiffc'
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1379): error C2672: 'run_absdiffc': no matching overloaded function found
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1379): error C2780: 'void cv::gapi::fluid::run_absdiffc(cv::gapi::fluid::Buffer &,const cv::gapi::fluid::View &,const float [])': expects 3 arguments - 4 provided
1>C:\Users\akhakim\OneDrive - Intel Corporation\projects\opencv\opencv\modules\gapi\src\backends\fluid\gfluidcore.cpp(1231): note: see declaration of 'cv::gapi::fluid::run_absdiffc'

Reply:
UNARY_ macro required fix number of arguments in function which it calls.

Reply:
So either I'll create a new macro which will duplicate the UNARY_ or leave all as it is.

Reply:
No, Anna, it doesn't require. There is a code snippet above which is the definition of `UNARY_` - there are no fix number of arguments.
You can actually remove the last argument from `run_absdiffc()` (str. 1224), BUT also need to remove it from your `UNARY_` instances in `GAPI_FLUID_KERNEL(GFluidAbsDiffC, cv::gapi::core::GAbsDiffC, false)` at the same time! (strs. 1391-1393)
This is why compiler has been arguing - you have deleted the argument from the function, but not from the `UNARY_` instance

That's what I was asking:
```
template<typename DST, typename SRC>
static void run_absdiffc(Buffer &dst, const View &src, const float scalar[4])

...

        UNARY_(uchar , uchar , run_absdiffc, dst, src, scalar);
        UNARY_(ushort, ushort, run_absdiffc, dst, src, scalar);
        UNARY_( short,  short, run_absdiffc, dst, src, scalar);
```

This will work, I checked it a week ago and double-checked it now

Reply:
Done.